tutor/src/models/course/period.js

```diff
@@ -60,14 +61,32 @@ class CoursePeriod extends BaseModel {
     this.course.periods.push(this);
   }
 
-  @action async findOrCreateTeacherStudentRole() {
+  @action async getTeacherStudentRole() {
     let role = this.course.roles.find((r) => (
       r.isTeacherStudent && r.period_id == this.id
     ));
     if (!role) {
       const { data } = await this.createTeacherStudent();
       role = this.course.roles.find(r => r.id == data.id );
     }
+    runInAction(() => {
+      role.joined_at = new Date(); // adjust the date so it always appears new
+      let student = find(this.course.students, { id: Student.TEACHER_AS_STUDENT_ID });
+      if (!student) {
+        this.course.students.push({ id: Student.TEACHER_AS_STUDENT_ID });
+        student = last(this.course.students);
```

Comment:
This would be clearer like:
```
student = { id: Student.TEACHER_AS_STUDENT_ID };
this.course.students.push(student);
```

Unless push is doing some conversion magic, of course.

Reply:
yeah, push is indeed doing magic :D it converts the plain object to a instance of Student class